lib/composer.js

```diff
@@ -154,33 +153,80 @@ export default class Composer {
       return false
     }
 
-    const { rootFilePath, jobnames } = this.initializeBuild(filePath)
+    latex.log.group('LaTeX Clean')
+
+    const { builder, rootFilePath, jobnames } = this.initializeBuild(filePath)
+    if (!builder) return false
+
+    const jobs = jobnames.map(jobname => this.cleanJob(builder, rootFilePath, jobname))
 
-    const jobs = jobnames.map(jobname => this.cleanJob(rootFilePath, jobname))
+    await Promise.all(jobs)
 
-    return _.flatten(await Promise.all(jobs))
+    latex.log.groupEnd()
   }
 
-  async cleanJob (rootFilePath, jobname) {
-    let { dir, name } = path.parse(rootFilePath)
+  async cleanJob (builder, rootFilePath, jobname) {
+    const rootPath = path.dirname(rootFilePath)
+    const generatedFiles = this.getGeneratedFileList(builder, rootFilePath, jobname)
+    let files = new Set()
+
+    const patterns = this.getCleanPatterns(builder, rootFilePath, jobname)
+
+    for (const pattern of patterns) {
+      const absolutePattern = path.join(rootPath, pattern)
+      // If the original pattern is absolute then we use it as a globbing pattern
+      // after we join it to the root, otherwise we use it against the list of
+      // generated files.
+      if (pattern[0] === path.sep) {
+        for (const file of glob.sync(absolutePattern)) {
+          files.add(path.normalize(file))
+        }
+      } else {
+        for (const file of generatedFiles.values()) {
+          if (minimatch(file, absolutePattern)) {
+            files.add(file)
+          }
+        }
+      }
+    }
+
+    const fileNames = Array.from(files.values()).map(file => path.basename(file)).join(', ')
+    latex.log.info('Cleaned: ' + fileNames)
 
-    const outdir = atom.config.get('latex.outputDirectory')
-    if (outdir) {
-      dir = path.join(dir, outdir)
+    for (const file of files.values()) {
+      fs.removeSync(file)
     }
+  }
 
-    if (jobname) {
-      name = jobname
+  getCleanPatterns (builder, rootFilePath, jobname) {
+    const { name, ext } = path.parse(rootFilePath)
+    const output = atom.config.get('latex.outputDirectory')
```

Comment:
I really think we should change `output` to `outputDir`. This applies to all other instances of `output`.